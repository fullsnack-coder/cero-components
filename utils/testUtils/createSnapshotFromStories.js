import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
