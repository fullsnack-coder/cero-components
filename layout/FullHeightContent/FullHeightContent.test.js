import * as stories from './FullHeightContent.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ layout / FullHeightContent ]', () => {
  describe('#FullHeightContent', () =>
    createSnapshotFromStories('FullHeightContent', stories))
})
