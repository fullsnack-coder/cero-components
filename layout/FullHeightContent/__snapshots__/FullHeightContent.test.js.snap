// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ layout / FullHeightContent ] #FullHeightContent when 'FullHeightContent' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="full-height-content is-playground"
  >
    <div
      class="content"
    >
      Mice defiling required Bilbo sailing unspoiled this hunting. 14th refuge knowing breath Middle-earth sack demon children?
    </div>
    <div
      class="spacer vertical"
      style="height: 15px;"
    />
    <div
      class="footer"
    >
      Ambition living aid indeed mutton suffered l failing kindness Ã‰othain? Red others Goblin-town spot stuff riddles hero?
    </div>
  </div>
</DocumentFragment>
`;
